### DISCUSSION TITLE
Multiline event description

### DISCUSSION THREAD

**Discussion by eguibs** [nhn/tui.calendar#1316] [General]
Hello, i would like to do a multiline event in the calendar as suggested by [sriramramaswami](https://github.com/sriramramaswami)
in this topic https://github.com/nhn/tui.calendar/issues/1078

Like illustrated with this picture :
[https://user-images.githubusercontent.com/8266445/162788443-7fe214a3-4f56-40e4-87b5-83a942f43558.png
](url)

I'm looking to do multiline event description. 

I know that its not planned to address this problem, but i would like to know if i can do myself a workaround.

do you know where this height is defined (hardcoded into the JS or on the CSS maybe ?? 
if i make something interesting i would propose it as a merge request :) 

Thanks  !!

**Comment by adhrinae** [comment]
Still, It's technically very hard to implement.

I'll explain a bit of code to you. You need to be familiar with [Preact](https://preactjs.com/).

https://github.com/nhn/tui.calendar/blob/29a095d3536d50eea5abcfac03d160ea7cf5f0f4/apps/calendar/src/components/dayGridWeek/alldayGridRow.tsx#L74-L83

It's rendering all-day events by calculating the absolute height of each event. and the height is fixed (22px, `EVENT_HEIGHT`).

`HorizontalEvent` receives its height by the `eventHeight` prop. and it will be the inline style of the dom directly.

https://github.com/nhn/tui.calendar/blob/29a095d3536d50eea5abcfac03d160ea7cf5f0f4/apps/calendar/src/components/events/horizontalEvent.tsx#L261
https://github.com/nhn/tui.calendar/blob/29a095d3536d50eea5abcfac03d160ea7cf5f0f4/apps/calendar/src/css/events/grid.css#L1-L9

The body of the event uses `white-space: nowrap` to handle overflowing text.

So you can change the CSS not to handle text overflow and remove fixed height. but removing fixed height will break the entire layout.
